lib/pages/chat_adaptive_scaffold/chat_adaptive_scaffold_builder.dart

```diff
@@ -0,0 +1,152 @@
+        MessageStyle.messageBubbleDesktopMaxWidth;
+    return ValueListenableBuilder(
+      valueListenable: rightColumnTypeNotifier,
+      builder: (context, rightColumnType, body) {
+        return Container(
+          color: Theme.of(context).colorScheme.surface,
+          child: AdaptiveLayout(
+            body: SlotLayout(
+              config: {
+                const WidthPlatformBreakpoint(
+                  end: breakpoint,
+                ): SlotLayout.from(
+                  key: AppAdaptiveScaffold.breakpointMobileKey,
+                  builder: (_) => Stack(
+                    children: [
+                      body!,
+                      if (rightColumnType.isShown)
+                        widget.rightBuilder(
+                          this,
+                          isInStack: true,
+                          type: rightColumnType,
+                        ),
+                    ],
+                  ),
+                ),
+                const WidthPlatformBreakpoint(
+                  begin: breakpoint,
+                ): SlotLayout.from(
+                  key: AppAdaptiveScaffold.breakpointWebAndDesktopKey,
+                  builder: (_) => Padding(
+                    padding: rightColumnType.isShown
+                        ? ChatAdaptiveScaffoldStyle.webPaddingRight
+                        : EdgeInsets.zero,
+                    child: ClipRRect(
+                      borderRadius: ChatAdaptiveScaffoldStyle.borderRadius,
+                      child: body!,
+                    ),
+                  ),
+                ),
+              },
+            ),
+            bodyRatio: 0.7,
+            internalAnimations: false,
+            secondaryBody: rightColumnType.isShown
+                ? SlotLayout(
+                    config: {
+                      const WidthPlatformBreakpoint(
+                        end: breakpoint,
+                      ): SlotLayout.from(
+                        key: AppAdaptiveScaffold.breakpointMobileKey,
+                        builder: null,
+                      ),
+                      const WidthPlatformBreakpoint(
+                        begin: breakpoint,
+                      ): SlotLayout.from(
+                        key: AppAdaptiveScaffold.breakpointWebAndDesktopKey,
+                        builder: (_) => ClipRRect(
+                          borderRadius: ChatAdaptiveScaffoldStyle.borderRadius,
+                          child: widget.rightBuilder(
+                            this,
+                            isInStack: false,
+                            type: rightColumnType,
+                          ),
+                        ),
+                      ),
+                    },
+                  )
+                : null,
+          ),
+        );
+      },
+      child: widget.bodyBuilder(this),
+    );
+  }
+}
+
+enum RightColumnType {
+  none,
+  search,
+  chatDirectDetails,
```

Comment:
I think it is not `chat direct details`, it is profile info. in the future, we will open the profile info also when use click on the avatar of user inside group chat.

Reply:
Agree with @hoangdat

Reply:
done, refactored from chat direct detail to profile info